### PR TITLE
Kernel: add packages for common USB DVB adapters (v5)

### PR DESCRIPTION
**Pull request by ghost**
Adds DVB drivers intended to be used with tvheadend package and easily available Astrometa DVB-C/T/T2 and Hauppauge soloHD/dualHD USB DVB tuners.
This allows users to record and stream TV programs from DVB networks to tvheadend clients such as Kodi. It allows watching home TV from abroad via VPN.
The new multimedia category makes it easy to add new tuners in the future.

Changes:
- Add packages for these kernel modules:
  - DVB core (installs dvb-core.ko)
  - DVB USB v2 (installs dvb-usb-v2.ko)
  - Drivers for chips used in Astrometa DVB:
    - Rafael Micro R820T (tuner)
    - Realtek RTL28xxU, RTL2830, RTL2832
    - Panasonic MN88472 (no firmware)
    - Panasonic MN88473 (no firmware)
    - Sony CXD2841ER
  - Drivers for chips used in Hauppauge soloHD/dualHD:
    - Silicon Labs Si2157
    - Silicon Labs Si2168 (no firmware)
    - Empia EM28xx (installs em28xx.ko, em28xx-dvb.ko and tveeprom.ko)
- Added new kernel config lines for v6.6

### CODE REVIEW

**Comment by motolav** [comment]
You shouldn't open new PRs for changes

**Comment by xback** [comment]
This will probably collide with my changes which I would push tomorrow.

https://git.openwrt.org/?p=openwrt/staging/xback.git;a=summary

**Comment by M95D** [comment]
Since I opened this pull request, more and more changes were needed to keep up with the main branch. I can't test it every single time. Please let me know if you decide to merge it.

File: package/kernel/linux/modules/multimedia.mk
Comment (zdz2019) [inline package/kernel/linux/modules/multimedia.mk:32]: Need to add:
$(LINUX_DIR)/drivers/media/rc/rc-core.ko

Comment (M95D) [reply package/kernel/linux/modules/multimedia.mk:32]: Did you encounter any error?

I know that Astrometa has an IR receiver, but remotes should be a separate package. Most users won't need a remote for their router.

**Comment by zdz2019** [comment]
The x86 test failed.

Collected errors:
 * pkg_hash_check_unresolved: cannot find dependency kmod-multimedia for kmod-usb-audio
 * pkg_hash_fetch_best_installation_candidate: Packages for kmod-usb-audio found, but incompatible with the architectures configured
 * opkg_install_cmd: Cannot install package kmod-usb-audio.

File: package/kernel/linux/modules/multimedia.mk
Comment (zdz2019) [reply package/kernel/linux/modules/multimedia.mk:32]: > Did you encounter any error?
> 
> I know that Astrometa has an IR receiver, but remotes should be a separate package. Most users won't need a remote for their router.

x86_64 compilation error, need rc-core.ko to compile successfully.

Comment (M95D) [reply package/kernel/linux/modules/multimedia.mk:32]: I can't reproduce. I need your .config to try it myself.

AFAIK, there's no dependency in kernel menuconfig between dvb and rc.
Also, your proposed change, `$(LINUX_DIR)/drivers/media/rc/rc-core.ko`, only installs the kernel module after it was successfully built. It doesn't make any sense.

**Comment by M95D** [comment]
It seems that there is an error with some imx targets.
`[...] v4l2-mem2mem.ko is missing.`
Can anyone help with a suggestion why this module isn't built and how to fix it? Thanks.

**Comment by jellekaufmann** [comment]
I have been using these patches on an ath79 archer c7 v2 for a couple of weeks now and it works fine in combination with minisatip :)

Would be nice to see it upstreamed

**Comment by M95D** [comment]
I split it into 3 patches to make it to review.

**Comment by M95D** [comment]
Fixed a recursive dependency. Ready to merge.

**Comment by M95D** [comment]
@neheb: May I ask for a review, please, if you have the time. This PR is more than a year old. Or simply reject it, so I can close and forget about it. Thank you!

**Review by neheb — APPROVED** [review]

**Comment by Shine-** [comment]
This would be a great addition to OpenWrt. I've been using Armbian for TVheadend so far, exactly because I couldn't get these Astrometa tuners to work in OpenWrt (or rather - I gave up early, since Armbian was working out-of-the-box anyway).  I suppose it wouldn't be hard to add more tuners once this is merged, based on your work (I have some Silicon Labs based ones in use as well)?

**Comment by jellekaufmann** [comment]
Agreed. This is a very useful addition.

It still works fine with my two dvb-c tuners.

**Comment by M95D** [comment]
Rebased.
@neheb, thanks for the review. How about a merge?

**Comment by neheb** [comment]
If I had merge access, sure

**Comment by M95D** [comment]
@hauke, please review & merge? Thank you!

**Comment by M95D** [comment]
Changes since v4:
- Removed mc.ko - it wasn't strictly needed. Without mc.ko, there's no USB audio dependency anymore.
- Added support for Hauppauge soloHD and dualHD tuners. PCTV Systems tripleStick T2 may work too, as they use the same chips.

**Comment by jneuhauser** [comment]
Thank you @M95D for not giving up this PR!

**Pull request closed without merging**
